private/buf/bufsync/syncer.go

```diff
@@ -34,15 +35,24 @@ type syncer struct {
 	repo                      git.Repository
 	storageGitProvider        storagegit.Provider
 	errorHandler              ErrorHandler
-	modulesToSync             []Module
-	syncPointResolver         SyncPointResolver
 	syncedGitCommitChecker    SyncedGitCommitChecker
 	moduleDefaultBranchGetter ModuleDefaultBranchGetter
-	allBranches               bool
+	syncPointResolver         SyncPointResolver
+
+	// flags received on creation
+	sortedModulesDirsForSync []string
+	modulesDirsForSync       map[string]struct{}
+	syncAllBranches          bool
+
+	commitsToTags                   map[string][]string                               // commits:[]tags
+	branchesToModulesForSync        map[string]map[string]bufmoduleref.ModuleIdentity // branch:moduleDir:moduleIdentityInHEAD
```

Comment:
Used the suffix `*ForSync` instead of `*ToSync` in many variables in this pkg so it wouldn't be confused with the `To` that separates keys and values in maps, like in this one.